source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -384,27 +383,42 @@ void CacheFilter::processSuccessfulValidation(Http::ResponseHeaderMap& response_
     return Http::HeaderMap::Iterate::Continue;
   });
 
-  if (should_update_cached_entry) {
-    // TODO(yosrym93): else the cached entry should be deleted.
-    // Update metadata associated with the cached response. Right now this is only response_time;
-    const ResponseMetadata metadata = {time_source_.systemTime()};
-    cache_.updateHeaders(*lookup_, response_headers, metadata);
-  }
+  // Update metadata associated with the cached response. Right now this is only response_time;
+  const ResponseMetadata metadata = {time_source_.systemTime()};
+  cache_.updateHeaders(*lookup_, response_headers, metadata);
```

Comment:
In the old code, this call was gated by `should_update_cache_entry` whereas in the changed code this is unconditional. Is that because of the change on line 91 from `isResponseModified` to `isSuccessfulValidation`?

Reply:
That's correct. `shouldUpdateCacheEntry` and `isResponseNotModified` are combined into `isSuccessfulValidation` which gates the `updateHeaders` function itself. I made this change because 
1. Similar to the decision of inserting a cache entry, whether or not to update a header of an entry probably should be part of `CachePolicy`, which has all the logic that is required to evaluate both incoming and outgoing cache entries.
2. It makes sense that the step precedes `processSuccessfulValidation` is a check whether the validation is successful
3. If we call the `updateHeaders`, we expect the headers to be updated instead of unchanged because the response was not cachable.